libs/domain/picking/src/components/picking-product-card/picking-product-card.component.ts

```diff
@@ -0,0 +1,184 @@
+import { i18n } from '@spryker-oryx/utilities';
+import { html, LitElement, TemplateResult } from 'lit';
+import { ifDefined } from 'lit-html/directives/if-defined.js';
+import { property, state } from 'lit/decorators.js';
+import { when } from 'lit/directives/when.js';
+import {
+  EVENT_EDIT,
+  EVENT_SUBMIT,
+  ItemsFilters,
+  PickingListItem,
+  ProductItemPickedEvent,
+  SummaryInfo,
+} from '../../models';
+import { styles } from './picking-product-card.styles';
+
+export class PickingProductCardComponent extends LitElement {
+  static styles = styles;
+
+  @property() productItem?: PickingListItem;
+  @property() status?: string;
+
+  @state() isCorrectNumberOfPickedProvided = true;
+  @state() currentNumberOfPicked?: number;
+  @state() pickedDataEvent?: ProductItemPickedEvent;
+
+  protected summaryInfo: SummaryInfo | undefined;
+
+  protected onSubmit(e: SubmitEvent): void {
+    e.preventDefault();
+
+    this.currentNumberOfPicked =
+      this.currentNumberOfPicked ?? this.productItem?.numberOfPicked;
+
+    if (this.productItem?.product.id) {
```

Comment:
```
this.dispatchPickingEvents(EVENT_SUBMIT, {
        productId: this.productItem.product.id!,
        numberOfPicked: this.currentNumberOfPicked,
});
```

Reply:
Optional chain expressions can return undefined by design - using a non-null assertion is unsafe and wrong

Reply:
Yes, but in render function you are rendering form with submit handling only when `this.productItem` is truthy, that means it can't dispatch submit till `this.productItem` is falsy.

Reply:
I think for this case using non-assertion could be, like this:
`productId: this.productItem!.product.id`

The productItem should never be null